File: workspace/java/src/test/java/com/dslplatform/ocd/values/TestArrayOptUUIDValue.java
package com.dslplatform.ocd.values;

import com.dslplatform.client.Bootstrap;
import com.dslplatform.ocd.values.ArrayOptUUIDInValue.ArrayOptUUIDValue;
import com.dslplatform.patterns.ServiceLocator;
import org.junit.*;
import static org.junit.Assert.*;

public class TestArrayOptUUIDValue {
    private static ServiceLocator locator;

    @BeforeClass
    public static void setUpClass() throws Exception {
//        locator = Bootstrap.init(TestArrayOptUUIDValue.class.getResourceAsStream("dsl-project.ini"));
    }

    @AfterClass
    public static void tearDownClass() throws Exception {
//        locator.resolve(java.util.concurrent.ExecutorService.class).shutdown();
//        locator = null;
    }

    @Before
    public void setUp() throws Exception {
    }

    @After
    public void tearDown() throws Exception {
    }

    @Test
    public void testFieldType() throws NoSuchFieldException {
        assertEquals(
                java.util.UUID[].class,
                ArrayOptUUIDValue.class.getDeclaredField("arrayOptUUID").getGenericType());
    }

    @Test
    public void testGetterType() throws NoSuchMethodException {
        assertEquals(
                java.util.UUID[].class,
                ArrayOptUUIDValue.class.getMethod("getArrayOptUUID").getGenericReturnType());
    }

    @Test
    public void testSetterType() throws NoSuchMethodException {
        assertEquals(
                ArrayOptUUIDValue.class,
                ArrayOptUUIDValue.class.getMethod("setArrayOptUUID", java.util.UUID[].class).getReturnType());
    }

    @Test
    public void testDefaultPropertyValue() {
        assertArrayEquals(
                new java.util.UUID[0],
                new ArrayOptUUIDValue().getArrayOptUUID());
    }
}

File: workspace/scala/src/test/scala/com/dslplatform/ocd/values/TestArrayOptUUIDValue.scala
package com.dslplatform.ocd.values

import ArrayOptUUIDInValue.ArrayOptUUIDValue
import com.dslplatform.api.patterns.ServiceLocator
import org.scalatest._
import scala.reflect.runtime.universe._

class TestArrayOptUUIDValue extends SpecExtensions {
  def `Test property field type: Array[Option[java.util.UUID]]`() =
    checkType(
      typeOf[Array[Option[java.util.UUID]]]
    , typeOf[ArrayOptUUIDValue].member("arrayOptUUID": TermName).asMethod.returnType
    )

  def `Test property default value: Array.empty[Option[java.util.UUID]]`() {
    assert(ArrayOptUUIDValue().arrayOptUUID === Array.empty[Option[java.util.UUID]])
  }
}
